### PR TITLE
Add issues and pull requests to the current release project

### PR DESCRIPTION
**Pull request by smashwilson**
More GitHub Action fun :zap:

This one will run each time we assign an issue or open a pull request. It'll add the issue or pull request to the current [release project board](https://github.com/atom/github/projects/24) so we keep that as an accurate record of the work that's been done within each release cycle.

### CODE REVIEW

**Comment by smashwilson** [comment]
Tested with #2132. Looks like it added itself to the release project correctly :zap:

**Review by vanessayuenn — APPROVED** [review]
Excited to see how this will work! Something that would be neat is to add an action triggered by us unassigning issues, which will take the issue off the current project board. It's just an 💡 idea 💡  though.

File: .github/main.workflow [hunk 1]
```diff
@@ -4,7 +4,39 @@ workflow "GraphQL schema update" {
...
+  uses = "./actions/auto-sprint"
+  secrets = ["GITHUB_TOKEN"]
+}
+
+action "Consider pull request for release board" {
+  uses = "actions/bin/filter@master"
+  args = "action 'opened|merged|assigned|reopened'"
+}
+
+action "Add pull request to release board" {
```
[the remark this answers is not in the document]
Comment (smashwilson) [reply .github/main.workflow]: Basically, "consider" filters on the event's action to determine whether or not it should result in a release board addition; "add" does the actual addition.

File: actions/auto-sprint/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  if (!teamResponse.organization.team.members.nodes.some(node => node.login === actorLogin)) {
+    tools.exit.neutral('User %s is not in the github-package team. Thanks for your contribution!', actorLogin);
+  }
+
+  // Identify the active release board and its "In progress" column
+  const projectQuery = await tools.github.graphql(`
+    query {
+      repository(owner: "atom", name: "github") {
+        projects(
+          search: "Release"
```
[the remark this answers is not in the document]
Comment (smashwilson) [reply actions/auto-sprint/index.js]: The search _is_ only on the name of the board:

_https://developer.github.com/v4/object/repository/, search for "projects"_

> Query to search projects by, currently only searching by name.

I'm not sure if it's case sensitive or not, though? The one I tested with was an exact case match 🤔

**Pull request merged**